mozjs/src/conversions.rs

```diff
@@ -536,7 +542,21 @@ pub unsafe fn latin1_to_string(cx: *mut JSContext, s: *mut JSString) -> String {
 
     let chars = slice::from_raw_parts(chars, length as usize);
     let mut s = String::with_capacity(length as usize);
-    s.extend(chars.iter().map(|&c| c as char));
+    let v = s.as_mut_vec();
+    for c in chars {
+        // This is safe as we are ensured that we only have "Latin1" strings which have the upper
+        // code word 00.
+        let i = std::mem::transmute::<_, AsciiChar>(*c).0 as u8;
+        v.push(i);
+    }
```

Comment:
Could this also be written with `extend`?

```suggestion
    s.extend(
        chars
        .iter()
        .map(|&char| std::mem::transmute::<_, AsciiChar>(*char).0 as u8),
    );
```

Reply:
Currently extend does not work with u8. Transforming `as u8 as char` shows a much slower time.